examples/systemcalls/systemcalls.c

```diff
@@ -58,12 +92,42 @@ bool do_exec(int count, ...)
  *   as second argument to the execv() command.
  *
 */
+    // Fork a child process
+    pid_t pid = fork();
 
-    va_end(args);
+    if (pid == -1)
+    {
+        perror("fork");
+        return false; // Fork failed
+    }
+    else if (pid == 0)
+    {
+        // Child process
+        execv(command[0], command);
+        perror("execv"); // This line is reached only if execv fails
+        exit(EXIT_FAILURE);
+    }
+    else
+    {
+        // Parent process
+        int status;
+        if (waitpid(pid, &status, 0) == -1)
+        {
+            perror("waitpid");
+            return false; // waitpid failed
+        }
 
-    return true;
+        // Check if the child process terminated successfully
+        if (WIFEXITED(status) && WEXITSTATUS(status) == 0)
+        {
+            return true; // Executed successfully
+        }
+        else
+        {
+            return false; // Command returned a non-zero exit status
+        }
+    }
```

Comment:
@aneesh1298 suggest finding a way to have a return path at the end of the function for readability.  Easiest option is probably to invert the logic on `!WIFEXITED(status) || WEXITSTATUS(status) != 0)` and handle the failure case there, handle `return true` at the end of the function.